src/sentry/taskworker/config.py

```diff
@@ -75,10 +75,10 @@ def wrapped(func):
         return wrapped
 
     def retry_task(self, taskdata: PendingTasks) -> None:
-        task_message = taskdata.to_message()
+        message = taskdata.to_proto()
         self.producer.produce(
             ArroyoTopic(name=self.topic),
-            KafkaPayload(key=None, value=task_message.SerializeToString(), headers=[]),
+            KafkaPayload(key=None, value=message.work.SerializeToString(), headers=[]),
         )
```

Comment:
We might want to move this to `pending_task_store.py` because it is using `PendingTasks` which is the Django model which should be kept in `pending_task_store.py`. Also the only thing that calls this is `pending_task_store.py`

Reply:
Having this accept a `Task` sounds like a good plan to me.